### ISSUE TITLE
clear/remove/hide/show

### ISSUE THREAD

**Issue by Schwad** [scarpe-team/scarpe#100]
Documenting this as it's on my mind.

```md
## Common Methods

A few methods are shared by every little element in Shoes.  Moving, showing,
hiding.  Removing an element.  Basic and very general things.  This list
encompasses those common commands.
```
`hide`, `remove` and `show` are methods that I think should be available to all widgets. I think? `remove` seems to actually "remove" what it's called on altogether.

Then `clear` clears everything in the slot. The same as calling `remove` on everything. It accepts an optional block that shows what we replace it with.

These could all be different issues but capturing this context _here_ as I think it could be a high value next addition to our API

**Comment by noahgibbs** [comment]
Clear, remove and append are now working. Still need to do hide/show.

**Comment by noahgibbs** [comment]
Also there's a "hidden" style in Shoes Classic, which is probably part of how I should implement hide/show.

**Comment by noahgibbs** [comment]
This will be fixed by #343 when it's merged.
